### PR TITLE
Raise errors for unsupported operations on certain types

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
Fixes: https://github.com/rapidsai/cudf/issues/15668

This PR raises errors for groupby operations on un-supported types.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/_lib/groupby.pyx [hunk 1]
```diff
@@ -167,6 +167,46 @@ cdef class GroupBy:
             included_aggregations_i = []
             col_aggregations = []
             for agg in aggs:
+                str_agg = str(agg)
```
Comment (mroeschke) [inline python/cudf/cudf/_lib/groupby.pyx:170]: Question: Is it not enough to raise the `TypeError` if the `if valid_aggregations == "ALL" or agg_obj.kind in valid_aggregations:` condition below fails?

Comment (galipremsagar) [reply python/cudf/cudf/_lib/groupby.pyx:170]: Ideally yes, but that is a very loose condition for many operations vs dtype matrix we have and libcudf seems to be silently returning empty columns for those. We need to dig deeper for other types aswell.

Comment (mroeschke) [reply python/cudf/cudf/_lib/groupby.pyx:170]: OK. I suppose this PR is a good enough intermediate solution until we can centeralize this in `get_valid_aggregation` or similar.

**Review by mroeschke — APPROVED** [review]

**Pull request merged**